### PR TITLE
Expose initial_pool_size and release_threshold parameters for cuda_async_memory_resource in Cython

### PR DESCRIPTION
**Pull request by shwina**
Merge after #752

### CODE REVIEW

File: python/rmm/_lib/memory_resource.pyx [hunk 1]
```diff
@@ -431,23 +457,8 @@ cdef class LoggingResourceAdaptor(UpstreamResourceAdaptor):
...
-            raise KeyError(key)
-        else:
-            ret = self.default_factory(key)
-            self[key] = ret
-            return ret
-
-
 # Global per-device memory resources; dict of int:DeviceMemoryResource
-cdef _per_device_mrs = KeyInitializedDefaultDict(CudaMemoryResource)
+cdef _per_device_mrs = defaultdict(CudaMemoryResource)
```
Comment (shwina) [inline python/rmm/_lib/memory_resource.pyx:461]: @kkraus14 @harrism : any idea why we were doing this? `CudaMemoryResource` was being passed an unused `device` arg,

File: python/rmm/_lib/memory_resource.pyx [hunk 2]
```diff
@@ -144,11 +145,36 @@ cdef class CudaMemoryResource(DeviceMemoryResource):
 cdef class CudaAsyncMemoryResource(DeviceMemoryResource):
     """
     Memory resource that uses cudaMallocAsync/Free for
-    allocation/deallocation
+    allocation/deallocation.
+
+    Parameters
+    ----------
+    initial_pool_size: int, optional
+        Initial size in bytes of the pool. If no value is provided,
```
Comment (rongou) [inline python/rmm/_lib/memory_resource.pyx:153]: Initial size of the pool in bytes?

**Review by rongou — APPROVED** [review]

File: python/rmm/_lib/memory_resource.pyx (hunk 1 above)
Comment (kkraus14) [reply python/rmm/_lib/memory_resource.pyx:461]: Not sure, but happy to see it cleaned up?

Comment (harrism) [reply python/rmm/_lib/memory_resource.pyx:461]: I have a vague memory of some problem we had to work around, but can't remember now...

File: python/rmm/_lib/memory_resource.pyx [hunk 3]
```diff
@@ -144,11 +145,36 @@ cdef class CudaMemoryResource(DeviceMemoryResource):
...
     """
     Memory resource that uses cudaMallocAsync/Free for
-    allocation/deallocation
+    allocation/deallocation.
+
+    Parameters
+    ----------
+    initial_pool_size: int, optional
+        Initial size of the pool in bytes. If no value is provided,
+        the initial size is 0.
```
Comment (jrhemstad) [inline python/rmm/_lib/memory_resource.pyx:154]: I changed this behavior here https://github.com/rapidsai/rmm/pull/757 to make it more consistent with the other pool.

Comment (shwina) [reply python/rmm/_lib/memory_resource.pyx]: Fixed docs to reflect this

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
@gpucibot merge

**Pull request merged**